core/shared/src/main/scala/zio/concurrent/ConcurrentSet.scala

```diff
@@ -110,10 +110,10 @@ final class ConcurrentSet[A] private (private val underlying: ConcurrentHashMap.
     UIO(underlying.asScala.toSet: @silent("JavaConverters"))
 
   @silent("JavaConverters")
-  def transform(f: A => A): UIO[Unit] = UIO {
+  def transform(f: A => A): UIO[Any] = UIO {
```

Comment:
I didn't touch the existing implementation as it's out of this PR scope, but I think we'll either have to make it atomic, or state the non-atomicity in its scaladoc.